instrumentation/jmx-metrics/library/src/main/java/io/opentelemetry/instrumentation/jmx/yaml/MetricStructure.java

```diff
@@ -91,47 +127,66 @@ protected void requireNonEmpty(String s, String msg) {
     }
   }
 
-  private static void addMetricAttributes(
-      List<MetricAttribute> list, Map<String, String> metricAttributeMap) {
-    if (metricAttributeMap != null) {
-      for (String key : metricAttributeMap.keySet()) {
-        String target = metricAttributeMap.get(key);
-        if (target == null) {
-          throw new IllegalStateException(
-              "nothing specified for metric attribute key '" + key + "'");
+  private List<MetricAttribute> addMetricAttributes(Map<String, String> metricAttributeMap) {
+
+    List<MetricAttribute> list = new ArrayList<>();
+    boolean hasStateAttribute = false;
+    for (String key : metricAttributeMap.keySet()) {
+      String target = metricAttributeMap.get(key);
+      if (target == null) {
+        throw new IllegalStateException("nothing specified for metric attribute key '" + key + "'");
+      }
+      MetricAttribute attribute = buildMetricAttribute(key, target.trim());
+      if (attribute.isStateAttribute()) {
+        if (hasStateAttribute) {
+          throw new IllegalArgumentException("only one state attribute is allowed");
         }
-        list.add(buildMetricAttribute(key, target.trim()));
+        hasStateAttribute = true;
       }
+      list.add(attribute);
+    }
+    if (hasStateAttribute && stateMapping.isEmpty()) {
```

Comment:
This might be fragile, as it assumes that `stateMapping` element will be always processed before `metricAttribute`. Also, `stateMapping` makes no sense if there's no `statekey` used (sorry if it is already checked and I do not see it).